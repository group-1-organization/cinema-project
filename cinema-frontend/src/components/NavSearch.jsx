import React from 'react';
import FormControl from 'react-bootstrap/FormControl'
import Form from 'react-bootstrap/esm/Form';
import Button from 'react-bootstrap/esm/Button';
import { Link } from "react-router-dom";
import { useState } from "react";
import '../Styles/NavSearch.css'

const NavSearch = () => {

    const [searchQuery, setSearchQuery] = useState("");
    //const [searchUrl, setSearchUrl] = useState("");

    const handleFilter = (e) => {
        //setSearchUrl(`/search/${searchQuery}`);
        setSearchQuery(`/search/${e.target.value}`);
    }

    return (
        <>
            <Form inline>
                <FormControl type="text" id="movieSearch" onInput={handleFilter} placeholder="Search Term" onFocus={(disappear) => disappear.target.placeholder = ''}
                    onBlur={(reappear) => reappear.target.placeholder = 'Search Term'} />

                <Link to={searchQuery} aria-label="search-link" ><Button aria-label="search-btn" className="button-colour adj-search" type="button" id="movieSearchSubmit" >Search Term</Button></Link>
            </Form>
        </>
    );
}

export default NavSearch;

//onChange={(event) => setSearchQuery(event.target.value)}
//onClick={handleFilter}