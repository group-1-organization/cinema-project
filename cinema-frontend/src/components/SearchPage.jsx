import { useParams } from "react-router";
import { Link } from "react-router-dom";
import { useEffect, useState } from "react";
import axios from 'axios';

const SearchPage = () => {

    let { query } = useParams();
    const [data, setData] = useState([]);

    useEffect(() => {
        axios.get(`http://localhost:5000/cinema/movies/find/${query}`)
            .then((res) => {
                const movieData = res.data;
                console.log(movieData);
                setData(movieData);
            }).catch((err) => {
                console.log(err.message);
                setData([]);
            })
    }, [query]);

    const pagePaths = ["Opening Times", "Getting There", "Classifications", "Places To Go", "About Us", "Contact Us", "Screens", "Listing", "Upcoming", "Discussion", "Booking"];
    let pageLinksCount = 0;

    const pageLinks = pagePaths.map(pagePath => {
        if (pagePath.toLowerCase().includes(query.toLowerCase())) {
            pageLinksCount++;
            return (
                <>
                    <Link to={`../${pagePath.replace(/\s/g, '')}`}><h1>{pagePath}</h1></Link>
                </>
            );
        }
    })

    const movieLinks = data.map(movie => {
        let relevantSearchTerms = []

        if (movie.title.toLowerCase().includes(query.toLowerCase())) {
            relevantSearchTerms.push(movie.title + " | ");
        }

        for (let director of movie.director) {
            if (director.toLowerCase().includes(query.toLowerCase())) {
                relevantSearchTerms.push(movie.director + " | ");
            }
        }

        for (let actor of movie.actors) {
            if (actor.toLowerCase().includes(query.toLowerCase())) {
                relevantSearchTerms.push(movie.actor + " | ");
            }
        }

        return (
            <>
                <Link to={`../mymovies/${movie.title}`}><h1>{movie.title}</h1></Link>
                <p>Relevant search terms: {relevantSearchTerms}</p>

            </>
        )
    });

    if (data.length || pageLinksCount > 0) {
        return (
            <>
                <h2>Search Results:</h2>
                {pageLinks}
                <br />
                <br />
                {movieLinks}
            </>
        )
    } else {
        return (
            <h2>No search results</h2>
        );
    }
}

export default SearchPage;