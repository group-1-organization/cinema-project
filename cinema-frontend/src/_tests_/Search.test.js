import renderer, { act } from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';

import SearchPage from '../components/SearchPage';
import NavSearch from '../components/NavSearch';

test('Search page matches snapshot', () => {
    const tree = renderer.create(<Router > <SearchPage /></Router >);

    expect(tree).toMatchSnapshot();
});

test('NavSearch matches snapshot', () => {
    const tree = renderer.create(<Router ><NavSearch /></Router >);

    expect(tree).toMatchSnapshot();
});