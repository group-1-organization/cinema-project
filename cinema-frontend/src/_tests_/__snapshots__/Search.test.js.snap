// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavSearch matches snapshot 1`] = `
<form
  className="form-inline"
>
  <input
    className="form-control"
    id="movieSearch"
    onBlur={[Function]}
    onFocus={[Function]}
    onInput={[Function]}
    placeholder="Search Term"
    type="text"
  />
  <a
    aria-label="search-link"
    href="/"
    onClick={[Function]}
  >
    <button
      aria-label="search-btn"
      className="button-colour adj-search btn btn-primary"
      disabled={false}
      id="movieSearchSubmit"
      type="button"
    >
      Search Term
    </button>
  </a>
</form>
`;

exports[`Search page matches snapshot 1`] = `
Array [
  " ",
  <h2>
    No matches found
  </h2>,
]
`;
